lib/trivial.nix

```diff
@@ -232,6 +232,31 @@ in {
 
   nixpkgsVersion = builtins.trace "`lib.nixpkgsVersion` is deprecated, use `lib.version` instead!" version;
 
+  /* The store path to this nixpkgs as a string with context. This can be used
+     to incur a dependency on this nixpkgs, for example, to symlink it into a
+     NixOS closure.
+
+     This is different than pkgs.path if nixpkgs is already in the Nix store,
+     since pkgs.path is a path to nixpkgs rather than a string-with-context,
+     and thus its usage typically, undesirably, results in copying this
+     nixpkgs.
+
+     See: https://github.com/NixOS/nix/issues/5868
+          https://github.com/NixOS/nix/issues/9428
+  */
+  nixpkgsStorePathString = if lib.trivial.inPureEvalMode || !(lib.strings.isStorePath (toString ../.))
```

Comment:
Can you add some tests for this function? In particular verifying that it returns a string with string context, you can probably use `builtins.getContext` for this. Either `lib/tests/misc.nix` or a `lib/tests/trivial.sh` file if Nix is not good enough to do the testing.

Reply:
Yep, done. It's hard to test the flake side automatically (perhaps we should have checks in the nixpkgs flake) but that has been tested manually and it is so trivial I can't imagine it breaking.